Bugzilla/API/V1/Github.pm

```diff
@@ -164,7 +171,167 @@ sub pull_request {
+  }
+
+  # Keep a list of bug ids that need to have comments added.
+  # We also use this for sending email later.
+  # Use a hash so we don't have duplicates. If multiple commits
+  # reference the same bug ID, then only one comment will be added
+  # with the text combined.
+  # When the comment is created, we will store the comment id to
+  # return to the caller.
+  my %update_bugs;
+
+  # Create a separate comment for each commit
+  foreach my $commit (@{$commits}) {
+    my $message = $commit->{message};
+    my $url     = $commit->{url};
+
+    if (!$url || !$message) {
+      return $self->code_error('github_pr_invalid_json');
+    }
+
+    # Find bug ID in the title and see if bug exists
+    my ($bug_id) = $message =~ /\b[Bb]ug[ -](\d+)\b/;
+    next if !$bug_id;
+
+    my $comment_text = "Authored by https://github.com/$pusher\n$url\n$message";
+
+    $update_bugs{$bug_id} ||= [];
+    push @{$update_bugs{$bug_id}}, {text => $comment_text};
+  }
+
+  # If no bugs were found, then we return an error
+  if (!keys %update_bugs) {
+    return $self->code_error('github_push_comment_bug_not_found');
+  }
+
+  # Set current user to automation so we can add comments to private bugs
+  my $auto_user = Bugzilla::User->check({name => 'automation@bmo.tld'});
+  $auto_user->{groups}       = [Bugzilla::Group->get_all];
+  $auto_user->{bless_groups} = [Bugzilla::Group->get_all];
+  Bugzilla->set_user($auto_user);
+
+  my $dbh = Bugzilla->dbh;
+  $dbh->bz_start_transaction;
+
+  # Actually create the comments in this loop
+  foreach my $bug_id (keys %update_bugs) {
+    my $bug = Bugzilla::Bug->new({id => $bug_id, cache => 1});
+    next if $bug->{error};
+
+    # Create a single comment if one or more commits reference the same bug
+    my $comment_text;
+    foreach my $comment (@{$update_bugs{$bug_id}}) {
+      $comment_text .= $comment->{text} . "\n\n";
+    }
+
+    $bug->add_comment($comment_text);
+
+    # If the bug does not have the keywork 'leave-open',
+    # we can also close the bug as RESOLVED/FIXED.
+    if (!$bug->has_keyword('leave-open')
+      && $bug->status ne 'RESOLVED'
+      && $bug->status ne 'VERIFIED')
+    {
+      # Set the bugs status to RESOLVED/FIXED
+      $bug->set_bug_status('RESOLVED', {resolution => 'FIXED'});
+
+      # Update the qe-verify flag if not set and the bug was closed.
+      my $found_flag;
+      foreach my $flag (@{$bug->flags}) {
+
+        # Ignore for all flags except `qe-verify`.
+        next if $flag->name ne 'qe-verify';
+        $found_flag = 1;
+        last;
+      }
+
+      if (!$found_flag) {
+        my $qe_flag = Bugzilla::FlagType->new({name => 'qe-verify'});
+        if ($qe_flag) {
+          $bug->set_flags(
+            [],
+            [{
+              flagtype => $qe_flag,
+              setter   => Bugzilla->user,
+              status   => '+',
+              type_id  => $qe_flag->id,
+            }]
+          );
+        }
+      }
```

Comment:
Setting a boolean flag and then running the code we want in a separate `if` seems unnecessary at first glance. We could eliminate the extra flag with the following change, any reason this wouldn't work?

```suggestion
        # Ignore for all flags except `qe-verify`.
        next if $flag->name ne 'qe-verify';

        my $qe_flag = Bugzilla::FlagType->new({name => 'qe-verify'});
        if ($qe_flag) {
          $bug->set_flags(
            [],
            [{
              flagtype => $qe_flag,
              setter   => Bugzilla->user,
              status   => '+',
              type_id  => $qe_flag->id,
            }]
          );
        }
        last;
      }
```

Reply:
That would work.

Reply:
> That would work.

Oh wait. The reason I did it that way is I only want to add the qe-verify flag if it is empty and not already set to either + or -.  So if $found_flag is true, then I don't want to do anything. So with your change I would be trying to add the new flag even if the flag is already set.